src/main/kotlin/com/exactpro/th2/lwdataprovider/RequestContext.kt

```diff
@@ -26,7 +26,9 @@ class RequestedMessageDetails(
     private val onResponse: ((RequestedMessageDetails) -> Unit)? = null
 ) {
     val id: String = storedMessage.id.toString()
-    val rawMessage: RawMessage = RawMessage.parseFrom(storedMessage.content)
+    val rawMessage: RawMessage = RawMessage.parseFrom(storedMessage.content).toBuilder()
+        .clearParentEventId() // we remove the parent event ID to avoid generating new events for the original parent
```

Comment:
What do you think about pass the current root or sub event to collect problem and mark owner of decode requests?

Reply:
I think it will be quite a big change right now. The RequestedMessageDetails class is been creating quite deep in the provider and passing the root event ID through the whole LDP does not look like a good idea right now